hack/make-rules/precheck-gopath.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Make sure the current directory is inside a GOPATH element.
+parts="$(echo "${GOPATH:-}" | sed 's/:/ /g')"
```

Comment:
I'd recommend exiting 1 if GOPATH is unbound. Otherwise the error message is a bit unclear (you have to build from a valid GOPATH and the current dir is in what you think you have for GOPATH, but it's empty (sudo invocation))